### PR TITLE
tiproxy: add tiproxyctl commands and http api

### PR DESCRIPTION
**Pull request by djshow832**
### What is changed, added or deleted? (Required)

More than one user asked about the TiProxy API: https://asktug.com/t/topic/1025865, https://asktug.com/t/topic/1024233/30

I didn't document it because it's only used for debugging and troubleshooting, but I found TiDB also documented it: https://docs.pingcap.com/zh/tidb/stable/tidb-monitoring-api

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: tiproxy/tiproxy-command-line-flags.md [hunk 1]
```diff
@@ -18,11 +18,29 @@ summary: 了解 TiProxy 的命令行参数。
 + 默认值：`""`
 + 必须指定配置文件。有关详细配置项，请参见[配置 TiProxy](/tiproxy/tiproxy-configuration.md)。注意，修改配置文件时 TiProxy 会自动重新加载配置，因此不要直接修改配置文件，建议通过 [`tiup cluster edit-config`](/tiup/tiup-component-cluster-edit-config.md) 或 [`kubectl edit tc`](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/modify-tidb-configuration) 修改配置。
 
-## TiProxy control
+## TiProxy Control
 
 本节介绍 TiProxy 客户端程序 `tiproxyctl` 的参数。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-command-line-flags.md:23]:
```suggestion
本节介绍 TiProxy 客户端程序 `tiproxyctl` 的语法、选项和命令。
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 2]
```diff
@@ -32,35 +50,67 @@ summary: 了解 TiProxy 的命令行参数。
...
 
 + 指定在与服务器建立连接时使用的证书。
 + 类型：`string`
 + 默认值：`""`
+
+### 命令
+
+#### `config set`
+
+`config set` 从标准输入读取 TOML 格式的配置，并将这些配置项设置到 TiProxy。其他未指定的配置项将保持不变，因此只需指定需要更改的配置项。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-command-line-flags.md:90]:
```suggestion
`tiproxyctl config set` 从标准输入读取 TOML 格式的配置，并将这些配置项设置到 TiProxy。其他未指定的配置项将保持不变，因此只需指定需要更改的配置项。
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 3]
```diff
@@ -32,35 +50,67 @@ summary: 了解 TiProxy 的命令行参数。
...
+level='warning'
+$ cat test.toml | tiproxyctl config set
+""
+$ tiproxyctl config get | grep level
+level = 'warning'
+```
+
+#### `config get`
+
+用于获取当前 TiProxy 的配置，输出格式为 TOML。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-command-line-flags.md:106]:
```suggestion
`tiproxyctl config get` 用于获取当前 TiProxy 的配置，输出格式为 TOML。
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 4]
```diff
@@ -32,35 +50,67 @@ summary: 了解 TiProxy 的命令行参数。
...
+
+#### `health`
+
+用于获取 TiProxy 的健康状况以及配置的校验和 (checksum)。当 TiProxy 正常运行时，返回配置的 checksum。当 TiProxy 正在关闭时，返回错误。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-command-line-flags.md:110]:
```suggestion
`tiproxyctl health` 用于获取 TiProxy 的健康状况以及配置的校验和 (checksum)。当 TiProxy 正常运行时，返回配置的 checksum。当 TiProxy 正在关闭时，返回错误。
```

**Review by Oreoxmt — APPROVED** [review]

File: tiproxy/tiproxy-command-line-flags.md [hunk 5]
```diff
@@ -32,35 +50,67 @@ summary: 了解 TiProxy 的命令行参数。
...
+level = 'warning'
+```
+
+#### `config get`
+
+`tiproxyctl config get` 用于获取当前 TiProxy 的配置，输出格式为 TOML。
+
+#### `health`
+
+`tiproxyctl health` 用于获取 TiProxy 的健康状况以及配置的校验和 (checksum)。当 TiProxy 正常运行时，返回配置的 checksum。当 TiProxy 正在关闭时，返回错误。
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:110]: 当 TiProxy 已经关闭时，也会返回错误吗

File: tiproxy/tiproxy-command-line-flags.md [hunk 6]
```diff
@@ -32,35 +50,67 @@ summary: 了解 TiProxy 的命令行参数。
...
 + 类型：`string`
 + 默认值：`""`
+
+### 命令
+
+#### `config set`
+
+`tiproxyctl config set` 从标准输入读取 TOML 格式的配置，并将这些配置项设置到 TiProxy。其他未指定的配置项将保持不变，因此只需指定需要更改的配置项。
+
+以下命令将 `log.level` 设置为 `"warning"`，其他配置项的值保持不变：
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:92]:
```suggestion
以下命令将 `log.level` 设置为 `'warning'`，其他配置项的值保持不变：
```

File: tiproxy/tiproxy-api.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+# TiProxy API
+
+[TiProxy](/tiproxy/tiproxy-overview.md) 提供 API 接口，用于获取其配置、健康状况以及监控数据等信息。
+
+> **注意：**
+>
+> TiProxy API 主要用于诊断调试，不保证与 TiProxy 未来引入的新特性完全兼容。因此不推荐客户在应用程序开发或工具开发中利用 TiProxy API 获取结果。
+
+TiProxy API 的地址为 `http://${host}:${port}`，其中 `host` 和 `port` 由 TiProxy 配置项 [`api.addr`](/tiproxy/tiproxy-configuration.md#addr-1) 指定。例如：
```
[the remark this answers is not in the document]
Comment (djshow832) [reply tiproxy/tiproxy-api.md:14]: Removed the link?

File: tiproxy/tiproxy-api.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+以下示例获取 JSON 格式的 TiProxy 配置：
+
+```bash
+curl "http://127.0.0.1:3080/api/admin/config/?format=json"
+```
+
+## 设置 TiProxy 的配置
+
+使用 TOML 格式修改 TiProxy 的配置。未指定的配置项将保持不变，因此只需指定需要更改的配置项。
```
[the remark this answers is not in the document]
Comment (djshow832) [reply tiproxy/tiproxy-api.md:42]: 不支持。可以的。

File: tiproxy/tiproxy-command-line-flags.md (hunk 5 above)
Comment (djshow832) [reply tiproxy/tiproxy-command-line-flags.md:110]: 是的

File: tiproxy/tiproxy-api.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (qiancai) [reply tiproxy/tiproxy-api.md:14]: added it back😂

File: tiproxy/tiproxy-api.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
```
Comment (qiancai) [inline tiproxy/tiproxy-api.md:42]:
```suggestion
目前，仅支持使用 TOML 格式修改 TiProxy 的配置。未指定的配置项将保持不变，因此只需指定需要更改的配置项。
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 10]
```diff
@@ -32,35 +50,67 @@ summary: 了解 TiProxy 的命令行参数。
...
+level = 'warning'
+```
+
+#### `config get`
+
+`tiproxyctl config get` 用于获取当前 TiProxy 的配置，输出格式为 TOML。
+
+#### `health`
+
+`tiproxyctl health` 用于获取 TiProxy 的健康状况以及配置的校验和 (checksum)。当 TiProxy 正常运行时，返回配置的 checksum。当 TiProxy 正在关闭时，返回错误。
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:110]:
```suggestion
`tiproxyctl health` 用于获取 TiProxy 的健康状况以及配置的校验和 (checksum)。当 TiProxy 正常运行时，返回配置的 checksum。当 TiProxy 处于关闭状态或者正在关闭时，返回错误。
```

File: tiproxy/tiproxy-api.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+    ```toml
+    [log]
+    encoder = 'tidb'
+    level = 'warning'
+    ```
+
+## 获取 TiProxy 的健康状况
+
+用于获取 TiProxy 的健康状况以及配置的校验和 (checksum)。当 TiProxy 正常运行时，返回配置的 checksum。当 TiProxy 正在关闭时，返回错误。
```
Comment (qiancai) [inline tiproxy/tiproxy-api.md:100]:
```suggestion
用于获取 TiProxy 的健康状况以及配置的校验和 (checksum)。当 TiProxy 正常运行时，返回配置的 checksum。当 TiProxy 处于关闭状态或者正在关闭时，返回错误。
```

File: tiproxy/tiproxy-command-line-flags.md [hunk 12]
```diff
@@ -32,35 +50,67 @@ summary: 了解 TiProxy 的命令行参数。
...
 
 + 指定在与服务器建立连接时使用的证书。
 + 类型：`string`
 + 默认值：`""`
+
+### 命令
+
+#### `config set`
+
+`tiproxyctl config set` 从标准输入读取 TOML 格式的配置，并将这些配置项设置到 TiProxy。其他未指定的配置项将保持不变，因此只需指定需要更改的配置项。
```
Comment (qiancai) [inline tiproxy/tiproxy-command-line-flags.md:90]:
```suggestion
`tiproxyctl config set` 从标准输入读取 TOML 格式的配置文件，并将这些配置项设置到 TiProxy。其他未指定的配置项将保持不变，因此只需指定需要更改的配置项。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**